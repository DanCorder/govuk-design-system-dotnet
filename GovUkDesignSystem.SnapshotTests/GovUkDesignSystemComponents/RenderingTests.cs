using System;
using System.Threading.Tasks;
using ApprovalTests;
using ApprovalTests.Namers;
using GovUkDesignSystem.GovUkDesignSystemComponents;
using GovUkDesignSystem.SnapshotTests.Helpers;
using Microsoft.AspNetCore.Mvc.Razor;
using Microsoft.AspNetCore.Mvc.ViewFeatures;
using Xunit;

namespace GovUkDesignSystem.SnapshotTests
{
    public class RenderingTests
    {
        private readonly ComponentTestServerFixture _server;

        public RenderingTests()
        {
            _server = new ComponentTestServerFixture();
        }

        [Theory]
        [InlineData("Hint", typeof(HintViewModel))]
        [InlineData("Label", typeof(LabelViewModel))]
        public async Task Rendering(string viewName, Type viewModelType)
        {
            // Tell ApprovalTests how to name these test results
            NamerFactory.AdditionalInformation = viewName;

            // Arrange
            var renderer = GetViewRenderer();
            var viewModel = Activator.CreateInstance(viewModelType);

            // Act
            var result = await renderer.Render(viewName, viewModel);

            // Assert
            Approvals.VerifyHtml(result);
        }

        private ViewRenderer GetViewRenderer()
        {
            var serviceProvider = _server.GetRequiredService<IServiceProvider>();
            var viewEngine = _server.GetRequiredService<IRazorViewEngine>();
            var tempDataProvider = _server.GetRequiredService<ITempDataProvider>();

            var viewRenderer = new ViewRenderer(viewEngine, tempDataProvider, serviceProvider);
            return viewRenderer;
        }
    }
}
