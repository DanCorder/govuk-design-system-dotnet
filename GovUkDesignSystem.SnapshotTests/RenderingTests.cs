using System;
using System.Threading.Tasks;
using GovUkDesignSystem.GovUkDesignSystemComponents;
using GovUkDesignSystem.SnapshotTests.Helpers;
using Microsoft.AspNetCore.Mvc.Razor;
using Microsoft.AspNetCore.Mvc.ViewFeatures;
using Xunit;

namespace GovUkDesignSystem.SnapshotTests
{
    public class RenderingTests
    {
        private readonly ComponentTestServerFixture _server;

        public RenderingTests()
        {
            _server = new ComponentTestServerFixture();
        }

        [Fact]
        public async Task Rendering_Label_ExpectSuccess()
        {
            // Arrange
            var renderer = GetViewRenderer();
            var viewModel = new LabelViewModel();

            // Act
            var result = await renderer.Render(@"Label", viewModel);

            // Assert
            Assert.NotNull(result);
        }

        private ViewRenderer GetViewRenderer()
        {
            var serviceProvider = _server.GetRequiredService<IServiceProvider>();
            var viewEngine = _server.GetRequiredService<IRazorViewEngine>();
            var tempDataProvider = _server.GetRequiredService<ITempDataProvider>();

            var viewRenderer = new ViewRenderer(viewEngine, tempDataProvider, serviceProvider);
            return viewRenderer;
        }
    }
}
