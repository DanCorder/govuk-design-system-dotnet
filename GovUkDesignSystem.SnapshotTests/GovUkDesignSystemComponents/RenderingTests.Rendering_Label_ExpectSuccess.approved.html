




            <label class="govuk-label"
                   
                   for=>

            </label>
         